test/Microsoft.AspNetCore.Razor.Language.Test/Legacy/HtmlTagsTest.cs

```diff
@@ -1,206 +1,180 @@
-                    new MarkupTagBlock(
-                        Factory.MarkupTransition("</text>"))));
+            ParseBlockTest("<text>Foo</text>}");
         }
 
         [Fact]
         public void CDataTag()
         {
-            ParseBlockTest("<![CDATA[Foo]]> Bar",
-                new MarkupBlock(
-                    Factory.Markup("<![CDATA[Foo]]>").Accepts(AcceptedCharactersInternal.None),
-                    Factory.Markup(" ").Accepts(AcceptedCharactersInternal.None)));
+            ParseBlockTest("<![CDATA[Foo]]> Bar");
         }
 
         [Fact]
         public void ScriptTag()
         {
-            ParseDocumentTest("<script>foo < bar && quantity.toString() !== orderQty.val()</script>",
-                new MarkupBlock(
-                    BlockFactory.MarkupTagBlock("<script>"),
-                    Factory.Markup("foo < bar && quantity.toString() !== orderQty.val()"),
-                    BlockFactory.MarkupTagBlock("</script>")));
+            ParseDocumentTest("<script>foo < bar && quantity.toString() !== orderQty.val()</script>");
         }
 
         [Fact]
         public void ScriptTag_WithNestedMalformedTag()
         {
-            ParseDocumentTest("<script>var four = 4; /* </ */</script>",
-                new MarkupBlock(
-                    BlockFactory.MarkupTagBlock("<script>"),
-                    Factory.Markup("var four = 4; /* </ */"),
-                    BlockFactory.MarkupTagBlock("</script>")));
+            ParseDocumentTest("<script>var four = 4; /* </ */</script>");
         }
 
         [Fact]
         public void ScriptTag_WithNestedEndTag()
         {
-            ParseDocumentTest("<script></p></script>",
-                new MarkupBlock(
-                    BlockFactory.MarkupTagBlock("<script>"),
-                    Factory.Markup("</p>"),
-                    BlockFactory.MarkupTagBlock("</script>")));
+            ParseDocumentTest("<script></p></script>");
         }
 
         [Fact]
         public void ScriptTag_WithNestedBeginTag()
         {
-            ParseDocumentTest("<script><p></script>",
-                new MarkupBlock(
-                    BlockFactory.MarkupTagBlock("<script>"),
-                    Factory.Markup("<p>"),
-                    BlockFactory.MarkupTagBlock("</script>")));
+            ParseDocumentTest("<script><p></script>");
         }
 
         [Fact]
         public void ScriptTag_WithNestedTag()
         {
-            ParseDocumentTest("<script><p></p></script>",
-                new MarkupBlock(
-                    BlockFactory.MarkupTagBlock("<script>"),
-                    Factory.Markup("<p></p>"),
-                    BlockFactory.MarkupTagBlock("</script>")));
+            ParseDocumentTest("<script><p></p></script>");
         }
 
-        [Theory]
-        [MemberData(nameof(VoidElementNames))]
-        public void VoidElementFollowedByContent(string tagName)
+        [Fact]
+        public void VoidElementFollowedByContent()
```

Comment:
Converted all these void element tests to be one large test all utilizing the same set of void elements.